src/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -0,0 +1,366 @@
+                    } else {
+                        nationSpawnPermission = NationSpawnLevel.UNAFFILIATED;
+                    }
+                } else {
+                    nationSpawnPermission = NationSpawnLevel.UNAFFILIATED;
+                }
+            }
+
+            // Check the permissions
+        	if (!(isTownyAdmin || ((nationSpawnPermission == NationSpawnLevel.UNAFFILIATED) ? nation.isPublic() : nationSpawnPermission.hasPermissionNode(plugin, player, nation))))
+         		throw new TownyException(TownySettings.getLangString("msg_err_nation_not_public"));
+
+        	break;
+        default:
+        	break;
+        }
+
+		// Prevent spawn travel while in disallowed zones (if configured.)
+		if (!isTownyAdmin) {
+			List<String> disallowedZones = TownySettings.getDisallowedTownSpawnZones();
+
+			if (!disallowedZones.isEmpty()) {
+				String inTown;
+				try {
+					Location loc = plugin.getCache(player).getLastLocation();
+					inTown = TownyAPI.getInstance().getTownName(loc);
+				} catch (NullPointerException e) {
+					inTown = TownyAPI.getInstance().getTownName(player.getLocation());
+				}
+				
+				if (inTown == null && disallowedZones.contains("unclaimed"))
+					throw new TownyException(String.format(TownySettings.getLangString("msg_err_x_spawn_disallowed_from_x"), spawnType.getTypeName(), TownySettings.getLangString("msg_the_wilderness")));
+				if (inTown != null && resident.hasNation() && townyUniverse.getDataSource().getTown(inTown).hasNation()) {
+					Nation inNation = townyUniverse.getDataSource().getTown(inTown).getNation();
+					Nation playerNation = resident.getTown().getNation();
+					if (inNation.hasEnemy(playerNation) && disallowedZones.contains("enemy"))
+						throw new TownyException(String.format(TownySettings.getLangString("msg_err_x_spawn_disallowed_from_x"), spawnType.getTypeName(), TownySettings.getLangString("msg_enemy_areas")));
+					if (!inNation.hasAlly(playerNation) && !inNation.hasEnemy(playerNation) && disallowedZones.contains("neutral"))
+						throw new TownyException(String.format(TownySettings.getLangString("msg_err_x_spawn_disallowed_from_x"), spawnType.getTypeName(), TownySettings.getLangString("msg_neutral_towns")));
+				}
+			}
+		}
+
+		Double travelCost = 0.0;
+		String spawnPermission = null;
+		TownyEconomyObject payee = null;
+		// Figure out costs, payee and spawnPermmission slug for money.csv log.
+        switch (spawnType) {
+        case RESIDENT:        	
+        	// Taking whichever is smaller, the cost of the spawn price set by the town, or the cost set in the config (which is the maximum a town can set their spawncost to.)
+        	travelCost = Math.min(townSpawnPermission.getCost(town), townSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", townSpawnPermission);
+        	payee = town;
+        	break;        
+        case TOWN:

```

Comment:
Duplicate switch statement. Instead replace with something like:
```java
case RESIDENT:
case TOWN:
...
```

Reply:
They weren't duplicates initially, resident didnt have teh option to pay the lower town spawn cost. Adjusting.